here-naksha-lib-core/src/main/java/com/here/naksha/lib/core/models/storage/HeapCacheCursor.java

```diff
@@ -143,6 +162,7 @@ public boolean absolute(long position) {
   @Override
   public @NotNull FEATURE addFeature(@NotNull FEATURE feature) {
     inMemoryData.add(createCodec(feature));
+    restoreOrderPossible = false;
```

Comment:
Why not simply `originalFeaturesOrder = null`? Would release the memory for the map and have the same effect?